app/assets/javascripts/index/history.js

```diff
@@ -112,7 +116,10 @@ OSM.History = function (map) {
 
   function toggleChangesetHighlight(id, state) {
     changesetsLayer.toggleChangesetHighlight(id, state);
-    $("#changeset_" + id).toggleClass("selected", state);
+    $("#sidebar_content .changesets ol li").removeClass("selected");
+    if (state) {
+      $("#changeset_" + id).addClass("selected");
+    }
```

Comment:
Do we need a similar change inside the `toggleChangesetHighlight` call on the layer or is there something that stops that getting "stuck" in the same way?

Reply:
Highlights don't get stuck in the layer because everything is redrawn when changesets enter/leave the viewport during scrolling. But I changed it anyway to match the highlight logic in the sidebar. Now only one changeset can be highlighted. Previously it was possible to highlight several, but this wasn't used for anything.